selfservice/strategy/oidc/strategy_registration.go

```diff
@@ -245,7 +259,7 @@ func (s *Strategy) Register(w http.ResponseWriter, r *http.Request, f *registrat
 	return errors.WithStack(flow.ErrCompletedByStrategy)
 }
 
-func (s *Strategy) registrationToLogin(w http.ResponseWriter, r *http.Request, rf *registration.Flow, providerID string) (*login.Flow, error) {
+func (s *Strategy) registrationToLogin(w http.ResponseWriter, r *http.Request, rf *registration.Flow) (*login.Flow, error) {
```

Comment:
Does this method keep working with PKCE or do we somehow lose track of PKCE? To test it, you could create an e2e test:

1. For an PKCE oauth2 provider with pkce force
2. Sign in with account A which does not exit
3. Kratos asks to sign up
4. Sign up with account A
5. Be logged in
6. Sign out
7. Perform sign up with account A
8. Expect log in
9. Log in with account A
10. Be signed in

If you want to be extra sure add validation errors in between (e.g. incomplete profile data from oidc)

Reply:
All this should continue working as before, because it's orthogonal to PKCE from what I can see in the code.

Whenever we move between flows, the OAuth2 token exchange has already taken place, so all PKCE-related stuff is already done.

Essentially, we store the verifier directly before redirecting the end user, and verify it right after he comes back (during token exchange). There are no cross-flow (`registrationToLogin` or the other way around) shenanigans to get in the way, I don't think.